File: silver/10815/python/main.py

if (__name__ == "__main__"):
    n = int(input())
    card_n = list(map(int, input().split()))[:n]
    m = int(input())
    card_m = list(map(int, input().split()))[:m]

    bucket = [0] * m

    for i in range(m):
        card_m[i] = (i, card_m[i])

    card_n.sort()
    card_m.sort(key=lambda x : x[1])

    count = 0
    for data in card_m:
        while (count < n and card_n[count] < data[1]):
            count += 1
        if (count >= n): break
        if (card_n[count] == data[1]): bucket[data[0]] = 1
    print(" ".join(map(str, bucket)))

    

File: silver/10815/c++/main.cpp
#include <iostream>
#include <algorithm>
#include <utility>

bool comp(const std::pair<int, int>& a, const std::pair<int, int>& b)
{
    return a.second < b.second;
}


int main()
{
    int n, m;
    std::cin >> n;
    int card_n[n];

    for (int i = 0; i < n; i++) {
        std::cin >> card_n[i];
    }

    std::cin >> m;
    std::pair<int, int> card_m[m];
    int bucket[m];
    int count = 0;

    for (int i = 0; i < m; i++) {
        std::cin >> card_m[i].second;
        card_m[i].first = i;
        bucket[i] = 0;
    }

    std::sort(card_n, card_n + n);
    std::sort(card_m, card_m + m, comp);


    for (int i = 0; i < m; i++) {
        while (card_n[count] < card_m[i].second) {
            count++;
        }
        if (count >= n) {
            break;
        }
        if (card_n[count] == card_m[i].second) {
            bucket[card_m[i].first] = 1;
        }
    }

    for (int data : bucket) {
        std::cout << data << ' ';
    }


    return 0;
}